channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -539,7 +539,9 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControlsAI {
      */
     private _xhrSender(payload: string[], isAsync: boolean) {
         const xhr = new XMLHttpRequest();
-        xhr[DisabledPropertyName] = true;
+        if (typeof Object.isExtensible !== 'function' || (typeof Object.isExtensible === 'function' && Object.isExtensible(xhr))) {
```

Comment:
While this is the correct and safe way to do this check, we have taken the approach of creating helper functions and then whitelisting specific usage.

See CoreUtils.objKeys(); CoreUtils.objDefineAccessors() etc for the helpers (Please create a helper for this method)

And then in the tools\rollup-es3 folder there is the source code for the "es3Poly()" and "es3Check()" rollup methods, both of these use tokens and exceptions listed in the Es3Tokens.ts file.

Reply:
While we prefer the helper approach, you can actually recode this portion "Object.isExtensible(xhr)" as "Object["isExtensible"](xhr) and this would fool the Es3Check() to allow the code. Which is why the wording for the error "or add explicit check for existence." you would still need the additional check that isExtensibile exists so that we maintain ES3 compatibility, we pick this up during code review.

Reply:
Thanks for your feedback.  I'll look at how to do this with a helper.  But it sounds like with your previous comments that this could cause some kind of looping behavior.  If I add this helper method and continue this effort, is this going to be possible to get this merged?